bitbots_convenience_frames/src/convenience_frames.cpp

```diff
@@ -0,0 +1,159 @@
+#include <ros/ros.h>
+#include <sensor_msgs/Imu.h>
+#include <std_msgs/Char.h>
+#include <std_msgs/Time.h>
+#include <tf2_ros/transform_broadcaster.h>
+#include <tf2_ros/transform_listener.h>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+#include <tf2/utils.h>
+
+
+class ConvenienceFramesBroadcaster
+{
+public:
+    ConvenienceFramesBroadcaster();
+private:
+    geometry_msgs::TransformStamped tf;
+    tf2_ros::Buffer tfBuffer;
+    tf2_ros::TransformListener tfListener;
+    bool is_left_support, got_support_foot;
```

Comment:
If I see it correctly: `tf`, `tfBuffer`, `tfListener` and `is_left_support` didn't get default initialized by the constructor. To leave them in an undefined state is not recommended always initialize them e.g. by using default constructors or default member initializers.